posts/app.py

```diff
@@ -53,5 +53,17 @@ def get_all_posts_for_event(event_id):
     """Get all posts matching the event with the specified ID."""
 
 
-if __name__ == "__main__":
+def connect_to_mongodb():  # pragma: no cover
+    """Connect to MongoDB instance using env vars."""
+    mongodb_uri = os.environ.get("MONGODB_URI")
+    if mongodb_uri is None:
+        print("Alert: not able to find MONGODB_URI environmental variable, "
+              "no connection to MongoDB instance")
+        return None  # not able to find db config var
```

Comment:
Returning `None` rather than throwing an exception will mean that the `DB` global might be `None`. Do you expect the rest of the code to handle that case, or will it simply move the error to elsewhere in the program?

Reply:
The reason I made it like that is so that unit tests running in Travis won't crash when the environment variable isn't set (i.e. when it's hidden in a PR Travis build). In effect, this should only be None in that specific case, but in the event a user forgot to set it in the Cloud Run config or export it for local development for example, it prints out that it wasn't able to connect to Mongo and the rest of the code will crash when trying to access `DB`. That was the best compromise between cleanliness and verbosity I could find, but maybe there's a better way.

Reply:
One other option is to return an object that throws a clear error on attribute access. You can override `__getattribute__` like so:

```python
class Thrower(object):
  def __getattribute__(self):
    throw DBNotConnected()
```

(You can even declare this class in the `connect_to_mongodb` method if you want to hide it from prying eyes.)

Reply:
Done https://github.com/knative-portability/large-events/pull/9/commits/92f1740d2eb916d45ba152a9c91775d200768f30